pilot/pkg/model/context.go

```diff
@@ -355,6 +355,10 @@ type WatchedResource struct {
 	// For Delta Xds, all resources of the TypeUrl that a client has subscribed to.
 	ResourceNames []string
 
+	// Wildcard indicates the subscription is a wildcard subscription. This only applies to types that
```

Comment:
Not sure this should be client driven. Server may decide to send some entries or none. Which entries are sent will be based on some criteria - my proposal is:
- if mesh size < 1000 - send all 
- else send endpoints in same namespace, up to 1000 ( with future refinements).

CP may also decide to not send anything at all and do on-demand only

Reply:
(that's not a blocking comment - we can iterate )

Reply:
In non-wildcard subscription, the server is still allowed to push more resources than requested, so we can still do that. At the very least, for now the on-demand client support is not strong so its nice to have wildcard; if everything is good at on-demand in the future maybe we drop it. But it can still be nice for debug tools, etc to be able to subscribe to everything?